st2api/tests/unit/controllers/v1/test_executions_rbac.py

```diff
@@ -135,3 +138,36 @@ def test_get_all_limit_minus_one(self):
 
         resp = self.app.get('/v1/actionexecutions?limit=-1')
         self.assertEqual(resp.status_code, http_client.OK)
+
+    def test_get_respective_actions(self):
+        cfg.CONF.set_override(name='permission_isolation', override=True,
+                              group='rbac')
+        data = {
+            'action': 'wolfpack.action-1',
+            'parameters': {
+                'actionstr': 'foo'
+            }
+        }
+
+        # User with admin role assignment
+        user_db = self.users['admin']
+        self.use_user(user_db)
+
+        resp = self._do_post(data)
+        self.assertEqual(resp.status_code, http_client.CREATED)
+
+        resp1 = self.app.get('/v1/actionexecutions?limit=-1')
```

Comment:
I wonder how this test would look like with positive limit. I have a feeling that it will queries the page first and filters users later which means the endpoint could potentially reduce the page of no records. Lets add a test and if it happens, let's deal with that before we merge.

Reply:
I update my PR and there is a raw user filter when querying resources in mongo.  All entries will be suitable for every user, so I there a page of no records won't exist.